src/controller/OperationalCredentialsDelegate.h

```diff
@@ -81,6 +82,13 @@ class DLL_EXPORT OperationalCredentialsDelegate
      */
     virtual void SetFabricIdForNextNOCRequest(FabricId fabricId) {}
 
+    /**
+     *   This function sets the CAT values for which the next NOC Chain should be generated. This API is
+     *   not required to be implemented if the delegate implementation has other mechanisms to get the
+     *   CAT values.
+     */
+    virtual void SetCATValuesForNextNOCRequest(CATValues cats) {}
```

Comment:
I'm sorry, this does not make sense.  The decision about CATs should be made by the delegate implementation, not the core SDK calling calls into the delegate.  This is unlike the fabric id (which is really information for that delegate implementation in terms of which fabric it's dealing with) or the node id (which is not really great, but given the API surface of the device controller, where we _have_ to know what node id is involved so we can do our operational lookup... though even there we could parse it out of the cert instead of doing what we do now).

The way this should really work is:

1. chip-tool already has an `ExampleOperationalCredentialsIssuer *`.  It should just be setting things on _that_ as desired.  Other consumers of `ExampleOperationalCredentialsIssuer` can do likewise if they want.
2. `ExampleOperationalCredentialsIssuer` can have API on it for setting next CATs.  But that should not be a general `OperationalCredentialsDelegate` API.
3. There shouldn't be any CATs in commissioning params or changes to CHIPDeviceController.cpp here.

Reply:
Removed the virtual function. But I still have a question. chip-tool can only get its `ExampleOperationalCredentialsIssuer` by `GetCredentialIssuer()` in `examples/chip-tool/commands/example/ExampleCredentialIssuerCommands.h`. But this function return an `OperationalCredentialsDelegate *` instead of `ExampleOperationalCredentialsIssuer *`. Should I do a `reinterpret_cast` here? Or is there a better solution for it?

Reply:
Change the function to return `ExampleOperationalCredentialsIssuer *`.